projects/components/src/file-upload/file-upload.service.ts

```diff
@@ -72,7 +74,7 @@ export class FileUploadService {
 
           return undefined;
         }),
-        this.notificationService.withNotification(`File upload successful`, `File upload failed`),
+        this.showFileUploadNotification(showNotification, `File upload successful`, `File upload failed`),
```

Comment:
This is a bit weird. Probably rename the method to `maybeShowFileUploadNotification` or just dont call it when boolean is false